patches/minecraft/net/minecraft/tags/NetworkTagManager.java.patch

```diff
@@ -0,0 +1,28 @@
+--- a/net/minecraft/tags/NetworkTagManager.java
++++ b/net/minecraft/tags/NetworkTagManager.java
+@@ -8,9 +8,9 @@
+ import net.minecraft.resources.IResourceManagerReloadListener;
+ 
+ public class NetworkTagManager implements IResourceManagerReloadListener {
+-   private final NetworkTagCollection<Block> field_199719_a = new NetworkTagCollection<Block>(Block.field_149771_c, "tags/blocks", "block");
+-   private final NetworkTagCollection<Item> field_199720_b = new NetworkTagCollection<Item>(Item.field_150901_e, "tags/items", "item");
+-   private final NetworkTagCollection<Fluid> field_205705_c = new NetworkTagCollection<Fluid>(Fluid.field_205588_a, "tags/fluids", "fluid");
++   private final NetworkTagCollection<Block> field_199719_a = net.minecraftforge.common.tags.ForgeTagCollection.fromForgeRegistry(net.minecraftforge.registries.ForgeRegistries.BLOCKS, net.minecraftforge.common.tags.ForgeTagCollection.getDefaultLoadingLocation(net.minecraftforge.registries.GameData.BLOCKS),"block");
++   private final NetworkTagCollection<Item> field_199720_b = net.minecraftforge.common.tags.ForgeTagCollection.fromForgeRegistry(net.minecraftforge.registries.ForgeRegistries.ITEMS, net.minecraftforge.common.tags.ForgeTagCollection.getDefaultLoadingLocation(net.minecraftforge.registries.GameData.ITEMS),"item");
++   private final NetworkTagCollection<Fluid> field_205705_c = new net.minecraftforge.common.tags.ForgeTagCollection<>(Fluid.field_205588_a, net.minecraftforge.common.tags.ForgeTagCollection.getDefaultLoadingLocation(net.minecraftforge.common.tags.ForgeTagManager.FLUID_TAGS), "fluid");
+ 
+    public NetworkTagCollection<Block> func_199717_a() {
+       return this.field_199719_a;
+@@ -47,10 +47,8 @@
+    }
+ 
+    public static NetworkTagManager func_199714_b(PacketBuffer p_199714_0_) {
+-      NetworkTagManager networktagmanager = new NetworkTagManager();
+-      networktagmanager.func_199717_a().func_200043_b(p_199714_0_);
+-      networktagmanager.func_199715_b().func_200043_b(p_199714_0_);
+-      networktagmanager.func_205704_c().func_200043_b(p_199714_0_);
++      net.minecraftforge.common.tags.ForgeTagManager networktagmanager = new net.minecraftforge.common.tags.ForgeTagManager();
```

Comment:
Do not hijack the vanilla packets.

Reply:
Forge has a handshake system, and a bunch of other things are are sent during play time. Sending tags as seperate lists avoids the single massive packet issue. And should be fine as they are just swapped out in memory.
Basically stop tieing everything together as it'll easily explode the limit that S->C packets have.